pkg/services/govmomi/extra/config.go

```diff
@@ -39,35 +46,46 @@ func (e *Config) SetCustomVMXKeys(customKeys map[string]string) error {
 
 // SetCloudInitUserData sets the cloud init user data at the key
 // "guestinfo.userdata" as a base64-encoded string.
-func (e *Config) SetCloudInitUserData(data []byte) error {
+func (e *Config) SetCloudInitUserData(data []byte) {
+	e.setUserData(guestInfoCloudInitData, guestInfoCloudInitEncoding, data)
+}
+
+// SetCloudInitMetadata sets the cloud init metadata at the key
+// "guestinfo.metadata" as a base64-encoded string.
+func (e *Config) SetCloudInitMetadata(data []byte) error {
```

Comment:
the CAPV networking relies on cloud-init metadata. so there might be work to do to support ignition for the network

Reply:
@yastij I think Flatcar/CoreOS networking relies on custom guestinfo case rather than metadata - https://kinvolk.io/docs/flatcar-container-linux/latest/installing/cloud/vmware/

Reply:
For ignition, we might want to set `guestinfo.hostname` to machine name, so the created node name matches the machine name(in Azure for example there a node/machine name match).

Reply:
if my understanding is corrrect reading above doc, ignition only recognize a single guestinfo.ignition.config.data, vs cloud-init we set guestinfo.userdata and guestinfo.metadata separately, where guestinfo.metadata contains network related settings like static IP. https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/pkg/util/constants.go#L19 and https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/pkg/services/govmomi/service.go#L217
So we may need to combine both userdata and metadata into one ignition config.

Reply:
@jzhoucliqr do you think this can be done in a follow-up PR?

Reply:
sounds good to me.